src/middlewared/middlewared/plugins/mdns.py

```diff
@@ -739,10 +739,9 @@ def __init__(self, **kwargs):
 
     def setup(self):
         webui = self.middleware.call_sync('datastore.query', 'system.settings')
-        if (webui[0]['stg_guiprotocol'] == 'http' or
-           webui[0]['stg_guiprotocol'] == 'httphttps'):
+        if not webui[0]['stg_guihttpsredirect']:
```

Comment:
Isnt HTTP always available? This should always be exposed, even if its a redirect.